rootfs/patcher-script.d/patch_wal_e_s3.py

```diff
@@ -0,0 +1,12 @@
+def patch_wal_e_hmac_auth_v4_handler():
```

Comment:
Do you have anything to read on why this patch is necessary? I am guessing this has to do with how S3 wants special authentication on every request `AWS4-HMAC-SHA256`. But doesn't latest boto version and latest wal-e already support this?

Reply:
```
# Avoid region_name is None.
# see: https://github.com/boto/boto/blob/master/boto/auth.py

def determine_region_name(self, host):
    parts = self.split_host_parts(host)
    if self.region_name is not None:
        region_name = self.region_name
    elif len(parts) > 1:
        if parts[1] == 'us-gov':
            region_name = 'us-gov-west-1'
        else:
            if len(parts) == 3:
                region_name = 'us-east-1'
            else:
                region_name = parts[1]
    else:
        region_name = parts[0]

    return region_name
```

see: [ this commit](https://github.com/teamhephy/boto/commit/88c980e56d1053892eb940d43a15a68af4ebb5e6)

Reply:
I see, makes sense. This was a hack in boto this whole time.

Reply:
Great work at eliminating boto as a dependency and appreciate your attention to detail and thoroughness. :1st_place_medal: